### PR TITLE
SStables: replace assertion with malformed_sstable_exception for invalid chunk_size

### LINKED ISSUE #20277: When SSTable compression metadata is corrupted, no hints of the underlying SSTable is shown in the logs
That's a result of assert usage, when doing a sanity check on chunk length in the compression metadata. So you'd need to either run a offline validate, or inspect the coredump. Let's improve it by switching to malformed exception instead.

### PR DESCRIPTION
**Pull request by Jager227**
This will allow to see underlying SStable file
Fixes #20277

### CODE REVIEW

File: sstables/compress.cc [hunk 1]
```diff
@@ -157,7 +157,9 @@ void compression::segmented_offsets::state::update_position_trackers(std::size_t
 }
 
 void compression::segmented_offsets::init(uint32_t chunk_size) {
-    SCYLLA_ASSERT(chunk_size != 0);
+    if (chunck_size == 0) {
```
Comment (denesb) [inline sstables/compress.cc:160]: typo: `chunck_size`

File: sstables/compress.cc [hunk 2]
```diff
@@ -157,7 +157,9 @@ void compression::segmented_offsets::state::update_position_trackers(std::size_t
...
+        throw sstables::malformed_sstable_exception("Segmented offsets chunk size is zero.");
```
Comment (denesb) [inline sstables/compress.cc:161]: Need to add sstable filename here.

Comment (Jager227) [reply sstables/compress.cc:161]: Output after corruption:
`Startup failed: std::runtime_error (Exception while populating keyspace 'system' with column family 'cdc_generations_v3' from datadir 'tmp/data/system/cdc_generations_v3-c697df3f55393ef5b003843be7ba1422': sstables::malformed_sstable_exception (Buffer improperly sized to hold requested data. Got: 151. Expected: 12336 in sstable tmp/data/system/cdc_generations_v3-c697df3f55393ef5b003843be7ba1422/me-3gk0_0xu7_34ycw248ycqx6sa8nv-big-CompressionInfo.db))
`

File: sstables/compress.cc (hunk 1 above)
Comment (raphaelsc) [reply sstables/compress.cc:160]: @Jager227 please fix the typo and rebase

**Comment by Jager227** [comment]
Rebased on latest master and fixed the typo

**Comment by denesb** [comment]
@raphaelsc please re-review.

File: sstables/compress.cc (hunk 2 above)
Comment (denesb) [reply sstables/compress.cc:161]: Turns out we already check this in the layer above, in sstables::parse().

**Pull request closed without merging**